port_drayage_plugin/src/port_drayage_plugin.cpp

```diff
@@ -13,290 +13,131 @@
+                                                              std::bind(&PortDrayageWorker::onRouteEvent, &pdw_, std_ph::_1));
+    georeference_subscriber_ = create_subscription<std_msgs::msg::String>("georeference", 1,
+                                                              std::bind(&PortDrayageWorker::onNewGeoreference, &pdw_, std_ph::_1));
+
+    // Setup publishers
+    outbound_mobility_operations_publisher_ = create_publisher<carma_v2x_msgs::msg::MobilityOperation>("outgoing_mobility_operation", 5);
+    ui_instructions_publisher_ = create_publisher<carma_msgs::msg::UIInstructions>("ui_instructions", 5);
+
+    // Setup service clients
+    set_active_route_client_ = create_client<carma_planning_msgs::srv::SetActiveRoute>("set_active_route");
+
+    // Set PortDrayageWorker variables
+    pdw_.setVehicleID(config_.cmv_id);
+    pdw_.setCargoID(config_.cargo_id);
+    pdw_.setEnablePortDrayageFlag(config_.enable_port_drayage);
+    pdw_.setStartingAtStagingAreaFlag(config_.starting_at_staging_area);
+
+    // Return success if everything initialized successfully
+    return CallbackReturn::SUCCESS;
+  }
+
+  bool PortDrayagePlugin::callSetActiveRouteClient(std::shared_ptr<carma_planning_msgs::srv::SetActiveRoute::Request> req)
+  {
+    // Send request to set_active_route service 
+    auto route_result = set_active_route_client_->async_send_request(req);
+
+    // Wait for response from service call
+    auto future_status = route_result.wait_for(std::chrono::milliseconds(100));
+
+    if (future_status == std::future_status::ready) {
+      if(route_result.get()->error_status == carma_planning_msgs::srv::SetActiveRoute::Response::NO_ERROR){
+        RCLCPP_DEBUG_STREAM(get_logger(), "Route Generation succeeded for Set Active Route service call.");
+        return true;
+      }
+      else{
+        RCLCPP_DEBUG_STREAM(get_logger(), "Route Generation failed for Set Active Route service call.");
+        return false;
+      }
     }
-
-    bool PortDrayagePlugin::call_set_active_route_client(cav_srvs::SetActiveRoute req){
-        if(_set_active_route_client.call(req)){
-            if(req.response.error_status == cav_srvs::SetActiveRouteResponse::NO_ERROR){
-                ROS_DEBUG_STREAM("Route Generation succeeded for Set Active Route service call.");
-                return true;
-            }
-            else{
-                ROS_DEBUG_STREAM("Route Generation failed for Set Active Route service call.");
-                return false;
-            }
-        }
-        else{
-            ROS_DEBUG_STREAM("Set Active Route service call was not successful.");
-            return false;
-        }
+    else {
+      // No response was received after making the service call
+      RCLCPP_DEBUG_STREAM(get_logger(), "Set Active Route service call was not successful.");
+      return false;
     }
 
-    bool PortDrayagePlugin::plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp){
```

Comment:
This (the maneuver planning functions here) was just dead code previously, correct?